bin/add_frontend_server.rb

```diff
@@ -67,41 +83,72 @@ def aws_instance_type(environment)
   end
 end
 
-# Returns a (instance_zone, frontend_name) typle for the given
-# ec2client.  The instance_zone is the one with least capacity amongst frontend instances,
-# and the frontend_name is one that is (probably) not currently already a known name to Chef,
-# except in infrequent race conditions.
+# Determine how the frontends are distributed across the zones
+def determine_frontend_instance_distribution
+  instances = @ec2client.describe_instances
 
-# param {string} ssh_username: The ssh username to use for connecting to the gateway.
-# param {frontend_name}: The base frontend name, or nil to use an automatically generated name.
-# (If the base name is not unique, a unique suffix will be added to it to provide uniqueness)
-def generate_instance_zone_and_name(ec2client, ssh_username, frontend_name = nil)
-  instances = ec2client.describe_instances
-
-  #Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
   frontend_instances = instances.reservations.map do |reservation|
-    reservation.instances.select{|instance| instance.state.name == 'running' && instance.tags.detect{|tag| tag.key ==
-                                                                                                     'Name' && tag.value.include?('frontend')}}
+    reservation.instances.select { |instance| instance.state.name == 'running' && instance.tags.detect { |tag| tag.key ==
+        'Name' && tag.value.include?('frontend') } }
   end
 
   frontend_instances.flatten!
 
-  instance_distribution = frontend_instances.each_with_object(Hash.new(0)){|(instance, _), instance_distribution|
-    instance_distribution[instance.placement.availability_zone] += 1}
-  determined_instance_zone, instance_count = instance_distribution.min_by{|_, v| v}
+  instance_distribution = frontend_instances.each_with_object(Hash.new(0)) { |(instance, _), instance_distribution|
+    instance_distribution[instance.placement.availability_zone] += 1 }
+
+  instance_distribution
+end
+
+# Return an array of objects that have the names and zones of instances to create
+#
+# When we are creating multiple frontends, don't bother trying to distribute in the underscaled instances. Just
+# distribute them across all of our existing zones
+def create_instance_provisioning_infos
+  instances_to_provision_array = []
+  instance_distribution = determine_frontend_instance_distribution
+  instances_to_provision = @options['count'].to_f
+
+  instances_to_provision_for_zones = Hash.new()
+
+  instance_distribution.keys.each do |zone|
+    instances_to_provision_for_zones[zone] = (instances_to_provision / instance_distribution.keys.count).floor
```

Comment:
If the arguments are integral this already does integral division with the floor.